nanome_rmsd/rmsd_selection.py

```diff
@@ -0,0 +1,379 @@
+from nanome.util import Logs
+from math import ceil
+from itertools import product
+
+# needleman wunsch algorithm
+def global_align(complex1,complex2,gap_penalty = -1, mismatch_penalty = -1, match_reward = 3):
+
+    selected_res1 = selected_res(complex1)
+    selected_res2 = selected_res(complex2)
+
+    # list of residues type of the complex
+    seq1 = list(map(lambda res: res.type, selected_res1))
+    seq2 = list(map(lambda res: res.type, selected_res2))
+
+    # run the "smart occupancy selection method" on the residue lists of both complexes
+    res_list1 =list(map(lambda a:select_occupancy(a),selected_res1))
+    res_list2 =list(map(lambda a:select_occupancy(a),selected_res2))
+    
+    # Logs.debug("res_list1 is ",res_list1)
+    Logs.debug("length of reslist1 is ",len(res_list1))
+    Logs.debug("length of reslist2 is ",len(res_list2))
+
+    # create the table of global alignment
+    m, n = len(seq1), len(seq2)
+    score = np.zeros((m+1, n+1))      
+    
+    # file the first column and first row of the table
+    for i in range(0, m + 1):
+        score[i][0] = gap_penalty * i
+    for j in range(0, n + 1):
+        score[0][j] = gap_penalty * j
+
+    # fill the table wtih scores
+    for i in range(1, m + 1):
+        for j in range(1, n + 1):
+            if seq1[i-1] == seq2[ j-1]:
+                match = score[i - 1][j - 1] + match_reward
+            else:
+                match = score[i - 1][j - 1] + mismatch_penalty
+            delete = score[i - 1][j] + gap_penalty
+            insert = score[i][j - 1] + gap_penalty
+            score[i][j] = max(match, delete, insert)
+    Logs.debug("table filled")
+    Logs.debug(score)
+    # Traceback and compute the alignment 
+    align1, align2 = '', ''
+    final1, final2 = '', ''
+    # start from the bottom right cell
+    i,j = m,n
+    while i > 0 and j > 0: 
+        score_current = score[i][j]
+        score_diagonal = score[i-1][j-1]
+        score_up = score[i][j-1]
+        score_left = score[i-1][j]
+        # two residuses match, only deselect when the selected atoms are not matched
+        if score_current == score_diagonal + match_reward and \
+           seq1[i-1] == seq2[j-1] and seq1[i-1] != 'UNK' and seq2[j-1] != 'UNK':
+
+            align1 += seq1[i-1]
+            align2 += seq2[j-1]
+            final1 += seq1[i-1]
+            final2 += seq2[j-1]
+            # Logs.debug(seq1[i-1],' ',seq2[j-1])
+            match1=list(map(lambda a:a.selected,res_list1[i-1].atoms))
+            match2=list(map(lambda a:a.selected,res_list2[j-1].atoms))
+            if match1 != match2:
+                Logs.debug("different atoms in the residues, please select backbone_only")
+                Logs.debug("The two residues are not same: ")
+                Logs.debug("type 1 is ",res_list1[i-1].type)
+                Logs.debug("type 2 is ",res_list2[j-1].type)
+                Logs.debug("atoms 1:   ",list(map(lambda a:a.name,res_list1[i-1].atoms)))
```

Comment:
Lots of heavy debug. Maybe add a "if Logs._is_verbose()" ? Sorry, it's still not a public API, but i need to make it public